develop/dev-guide-sample-application-nextjs.md

```diff
@@ -0,0 +1,213 @@
+---
+
+<!-- markdownlint-disable MD024 -->
+<!-- markdownlint-disable MD029 -->
+
+# Connect to TiDB by using mysql2 in Next.js
+
+TiDB is a MySQL-compatible database. [mysql2](https://www.npmjs.com/package/mysql2) is one of the most popular open-source Node.js drivers.
+
+This document shows you how to use TiDB and mysql2 to build a simple CRUD application.
+
+## Prerequisites
+
+- [Node.js **18**](https://nodejs.org/en/download/) or later.
+- [Git](https://git-scm.com/downloads).
+- TiDB cluster. If you don't have a TiDB cluster, you can create one by following the steps below:
+    - (Recommended) Refer to [Create a TiDB Serverless cluster](/develop/dev-guide-build-cluster-in-cloud.md#step-1-create-a-tidb-serverless-cluster) to create your own TiDB Cloud cluster.
+    - Refer to [Deploy a local test TiDB cluster](/quick-start-with-tidb.md#deploy-a-local-test-cluster) or [Deploy a formal TiDB cluster](/production-deployment-using-tiup.md) to create a local cluster.
+
+## Run the sample application
+
+This section shows you how to run the sample application code and connect to TiDB.
+
+### Step 1: Clone the sample application repository to your local machine
+
+```bash
+git clone git@github.com:tidb-samples/tidb-nextjs-vercel-quickstart.git
+```
+
+### Step 2: Install dependencies
+
+```bash
+cd tidb-nextjs-vercel-quickstart
+yarn
+```
+
+### step 3: Configure the connection string
+
+Depending on the way you deploy TiDB, use different methods to connect to the TiDB cluster.
+
+<SimpleTab>
+
+<div label="TiDB Serverless">
+
+1. In the TiDB Cloud Clusters page, select your TiDB Serverless cluster and go to the **Overview** page. Click **Connect** in the upper right corner.
+
+2. In the pop-up window, confirm that the configuration is consistent with your runtime environment.
+
+   - Endpoint Type is **Public**.
+   - Connect With is **General**.
+   - Operating System is your runtime environment.
+
+   <Tip>If you are running in Windows Subsystem for Linux (WSL), switch to the corresponding Linux distribution.</Tip>
+
+3. Click **Create Password** to generate a password.
+
+   <Tip>If you have generated a password before, you can use the original password directly, or click **Reset Password** to generate a new password.</Tip>
+
+4. Run the following command to copy and rename `.env.example` to `.env`:
+
+   ```bash
+   cp .env.example .env
+   ```
+
+5. Copy and paste the corresponding connection string to `.env`. The following is an example:
+
+   ```bash
+   TIDB_HOST='{gateway-region}.aws.tidbcloud.com'
+   TIDB_PORT='4000'
+   TIDB_USER='{prefix}.root'
+   TIDB_PASSWORD='{password}'
+   TIDB_DB_NAME='test'
+   ```
+
+   Replace the placeholders in `{}` with the values obtained in the **Connect** window.
+
+6. Save the file.
+
+</div>
+
+<div label="Self-hosted TiDB">
+
+1. Run the following command to copy and rename `.env.example` to `.env`:
+
+   ```bash
+   cp .env.example .env
```

Comment:
```suggestion
   cp .env.example .env
```
`cp` command may not work on windows.

https://superuser.com/questions/1709297/is-there-windows-copy-command-that-mimic-unix-cp-r-behavior